### PR TITLE
JBR-7456 add regression test checking if implementations of all available the random number generator algorithms can be instantiated  including the deafult one

### PR DESCRIPTION
**Pull request by vprovodin**

### CODE REVIEW

File: test/jdk/jb/java/util/RandomGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        Random.from(randomGenerator);
+        System.out.println("passed");
+
+    } catch (IllegalArgumentException e) {
+        errMsg = e.getMessage();
+        isFailed = true;
+    }
+}
+}
```
Comment (mkartashev) [inline test/jdk/jb/java/util/RandomGeneratorTest.java:54]: This is an extra `}` with no corresponding '{'

File: test/jdk/jb/java/util/RandomGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import java.util.Comparator;
```
Comment (mkartashev) [inline test/jdk/jb/java/util/RandomGeneratorTest.java:1]: It's best to have a copyright header even if we won't contribute this file to OpenJDK

File: test/jdk/jb/java/util/RandomGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ *          implementations of the random number generator algorithm including "L32X64MixRandom", which is default.
+ *          So it is expected that the test fails on the runtime because the default algorithm is not available. It
+ *          should pass on the JBRSDK binaries.
+ *          In jdk23 the `jdk.random` module has been removed from the JDK. All algorithm implementations have been
+ *          moved to the `java.base` module. So the test is valid for both kinds of binaries.
+ *
+ * @run main/othervm RandomGeneratorTest
+ */
+
+public static void main(final String[] args) throws Exception {
```
Comment (mkartashev) [inline test/jdk/jb/java/util/RandomGeneratorTest.java:19]: Implicitly declared classes require an extra compiler option; will jtreg work with this test?

File: test/jdk/jb/java/util/RandomGeneratorTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    System.out.println("testing all available RandomGenerator factories:");
+    RandomGeneratorFactory.all()
+            .forEach(RandomGeneratorTest::testRandomGenerator);
+
+    System.out.println();
+
+    if (isFailed)
```
Comment (mkartashev) [inline test/jdk/jb/java/util/RandomGeneratorTest.java:27]: There are many undeclared variables in the test such as `isFailed` and others.

File: test/jdk/jb/java/util/RandomGeneratorTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+static void testRandomGenerator(RandomGeneratorFactory<RandomGenerator> factory) {
+    try {
+        if (!wasDefaultTested) {
+            RandomGeneratorFactory<RandomGenerator> defaultFactory = RandomGeneratorFactory.getDefault();
+            wasDefaultTested = (defaultFactory.group().compareTo(factory.group()) == 0)
+                    && (defaultFactory.name().compareTo(factory.name()) == 0);
+        }
+
+        RandomGenerator randomGenerator = factory.create();System.out.printf("%s %s - ",
```
Comment (mkartashev) [inline test/jdk/jb/java/util/RandomGeneratorTest.java:42]: Need a line break after `;` in this line

**Comment by vprovodin** [comment]
Sorry, the test was developed&tested in one branch, but pr was submitted from another by mistake
All notes were fixed

File: test/jdk/jb/java/util/RandomGeneratorTest.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+    static void testRandomGenerator(RandomGeneratorFactory<RandomGenerator> factory) {
+        try {
+            if (!wasDefaultTested) {
+                RandomGeneratorFactory<RandomGenerator> defaultFactory = RandomGeneratorFactory.getDefault();
+                wasDefaultTested = (defaultFactory.group().compareTo(factory.group()) == 0)
+                        && (defaultFactory.name().compareTo(factory.name()) == 0);
+            }
+
+            RandomGenerator randomGenerator = factory.create();System.out.printf("%s %s - ",
```
Comment (mkartashev) [inline test/jdk/jb/java/util/RandomGeneratorTest.java:70]: Still need a line break after `;` in this line.

File: test/jdk/jb/java/util/RandomGeneratorTest.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024,  JetBrains s.r.o.
```
Comment (mkartashev) [inline test/jdk/jb/java/util/RandomGeneratorTest.java:2]: There's no need for `, ` in this line

**Review by mkartashev — CHANGES_REQUESTED** [review]

**Review by mkartashev — APPROVED** [review]

**Pull request merged**